### PR TITLE
Skip computing size of child objects

### PR DESCRIPTION
**Pull request by dfalbel**
This speeds up expanding objects with a lot of self-references for which it's very slow to compute the size of. 
Relates to https://github.com/posit-dev/positron/issues/4636
Built on top of https://github.com/posit-dev/ark/pull/629

### CODE REVIEW

**Review by lionel- — APPROVED** [review]
Was the performance problem because we were repeatedly computing the size for all nested objects?

I guess ideally we'd start with the leaves and compute sizes of intermediary nodes from the size of their components, this way the work we're doing is proportional to the size of the tree?

The apporach taken here seems like a reasonable workaround.

**Comment by dfalbel** [comment]
I didn't do a detailed benchmark but, when we expand the children, we return a list of `PositronVariables`, one for each child node. In the case of the big model in https://github.com/posit-dev/positron/issues/4636 many of the child nodes seem to include references to the `data.frame` like eg, the formula, and call objects. This causes the each child node to trigger a large computation for the size. 

Since we can't really tell which child object ultimatelly owns the actual data.frame, we would also probably `sum(size(child) for child in children)` much larger then the object size.

Eg if we try to size each element of a list built like:

```
x <- list(
  x1 = matrix(1, ncol = 10000, nrow = 10000)
)
for (l in letters) {
  x[[l]] <- x$x1
}
x
```

File: crates/ark/src/variables/variable.rs [hunk 1]
```diff
@@ -541,15 +541,15 @@ impl PositronVariable {
...
+                Self::from(display_name.clone(), display_name, object.sexp, true)
             },
         }
     }
 
     /**
      * Create a new Variable from an R object
      */
-    fn from(access_key: String, display_name: String, x: SEXP) -> Self {
+    fn from(access_key: String, display_name: String, x: SEXP, compute_size: bool) -> Self {
```
[the remark this answers is not in the document]
Comment (dfalbel) [reply crates/ark/src/variables/variable.rs:552]: Makes sense! I'll make this change

[the remark this answers is not in the document]
Comment (dfalbel) [reply crates/ark/src/variables/variable.rs:552]: Actually, the problem is that `PositronVariable` doesn't hold any reference to the RObject, so we can't `compute_size()` later. We can do something like 

```
var.size = object.size()
```

at the call site, what do you think of this approach?

[the remark this answers is not in the document]
Comment (lionel-) [reply crates/ark/src/variables/variable.rs:552]: yep sounds good

**Review by lionel- — APPROVED** [review]

**Pull request merged**